app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAddonsManagerRobot.kt

```diff
@@ -0,0 +1,163 @@
+                            withText(addonName)
+                        )
+                    )
+                )
+            )
+                .perform(click())
+
+            AddonDetailedMenuRobot().interact()
+            return AddonDetailedMenuRobot.Transition()
+        }
+    }
+
+    private fun installButtonForAddon(addonName: String) =
+        onView(
+            allOf(
+                withContentDescription(R.string.mozac_feature_addons_install_addon_content_description),
+                isDescendantOfA(withId(R.id.add_on_item)),
+                hasSibling(withChild(withText(addonName)))
+            )
+        )
+
+    private fun selectInstallAddon(addonName: String) {
+        mDevice.waitNotNull(
+            Until.findObject(By.textContains(addonName)),
+            waitingTime
+        )
+
+        installButtonForAddon(addonName)
+            .check(matches(isCompletelyDisplayed()))
+            .perform(click())
+    }
+
+    private fun assertAddonIsEnabled(addonName: String) {
+        installButtonForAddon(addonName)
+            .check(matches(not(isCompletelyDisplayed())))
+    }
+
+    private fun assertAddonPrompt(addonName: String) {
+        onView(allOf(withId(R.id.title), withText("Add $addonName?")))
+            .check(matches(isCompletelyDisplayed()))
+
+        onView(
+            allOf(
+                withId(R.id.permissions),
+                withText(containsString("It requires your permission to:"))
+            )
+        )
+            .check(matches(isCompletelyDisplayed()))
+
+        onView(allOf(withId(R.id.allow_button), withText("Add")))
+            .check(matches(isCompletelyDisplayed()))
+
+        onView(allOf(withId(R.id.deny_button), withText("Cancel")))
+            .check(matches(isCompletelyDisplayed()))
+    }
+
+    private fun assertDownloadingAddonPrompt() {
+        mDevice.waitNotNull(
+            Until.findObject(By.textContains("Successfully installed")),
+            waitingTime
+        )
+    }
+
+    private fun cancelInstall() {
+        onView(allOf(withId(R.id.deny_button), withText("Cancel")))
+            .check(matches(isCompletelyDisplayed()))
+            .perform(click())
+    }
+
+    private fun allowInstall() {
+        onView(allOf(withId(R.id.allow_button), withText("Add")))
+            .check(matches(isCompletelyDisplayed()))
+            .perform(click())
+    }
+
+    private fun assertAddonsItems() {
+        assertRecommendedTitleDisplayed()
+        assertAddons()
+    }
+
+    private fun assertRecommendedTitleDisplayed() {
+        onView(withText("Recommended"))
+            .check(matches(isCompletelyDisplayed()))
+    }
+
+    private fun assertEnabledTitleDisplayed() {
+        onView(withText("Enabled"))
+            .check(matches(isCompletelyDisplayed()))
+    }
+
+    private fun assertAddons() {
+        onView(
+            allOf(
+                isAssignableFrom(RelativeLayout::class.java),
```

Comment:
I'm not familiar with this, what does it assert?

Reply:
This checks if all of the elements in the ublock addon are present.